scrapy_zyte_api/providers.py

```diff
@@ -186,6 +191,17 @@ async def __call__(  # noqa: C901
                         }
                     )
                 continue
+            if cls_stripped in {CustomAttributes, CustomAttributesValues}:
+                zyte_api_meta["customAttributes"] = {
+                    k: (
+                        dict(v)
+                        if isinstance(v, frozenset)
+                        else list(v) if isinstance(v, tuple) else v
+                    )
```

Comment:
this code seems to be copy-pasted - time to add a helper function? Is it something like undo_make_hashable?

Reply:
Yeah.

Reply:
Done, and now I think both functions need tests. We don't actually test what we receive in mockserver requests.

Reply:
Makes sense, let's add a few tests!